nibabel/viewers.py

```diff
@@ -0,0 +1,473 @@
+
+    OrthoSlicer3d also supports 4-dimensional data, where multiple
+    3-dimensional volumes are stacked along the last axis.  For 4-dimensional
+    data the fourth figure axis can be used to control which 3-dimensional
+    volume is displayed.  Alternatively, the - key can be used to decrement the
+    displayed volume and the + or = keys can be used to increment it.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> a = np.sin(np.linspace(0,np.pi,20))
+    >>> b = np.sin(np.linspace(0,np.pi*5,20))
+    >>> data = np.outer(a,b)[..., np.newaxis]*a
+    >>> OrthoSlicer3D(data).show()  # doctest: +SKIP
+    """
+    # Skip doctest above b/c not all systems have mpl installed
+    def __init__(self, data, affine=None, axes=None, cmap='gray',
+                 pcnt_range=(1., 99.), figsize=(8, 8), title=None):
+        """
+        Parameters
+        ----------
+        data : ndarray
+            The data that will be displayed by the slicer. Should have 3+
+            dimensions.
+        affine : array-like | None
+            Affine transform for the data. This is used to determine
+            how the data should be sliced for plotting into the saggital,
+            coronal, and axial view axes. If None, identity is assumed.
+            The aspect ratio of the data are inferred from the affine
+            transform.
+        axes : tuple of mpl.Axes | None, optional
+            3 or 4 axes instances for the 3 slices plus volumes,
+            or None (default).
+        cmap : str | instance of cmap, optional
+            String or cmap instance specifying colormap.
+        pcnt_range : array-like, optional
+            Percentile range over which to scale image for display.
+        figsize : tuple
+            Figure size (in inches) to use if axes are None.
+        """
+        # Nest imports so that matplotlib.use() has the appropriate
+        # effect in testing
+        plt, _, _ = optional_package('matplotlib.pyplot')
+        mpl_img, _, _ = optional_package('matplotlib.image')
+        mpl_patch, _, _ = optional_package('matplotlib.patches')
+        self._title = title
+        self._closed = False
+
+        data = np.asanyarray(data)
+        if data.ndim < 3:
+            raise ValueError('data must have at least 3 dimensions')
+        if np.iscomplexobj(data):
+            raise TypeError("Complex data not supported")
+        affine = np.array(affine, float) if affine is not None else np.eye(4)
+        if affine.ndim != 2 or affine.shape != (4, 4):
+            raise ValueError('affine must be a 4x4 matrix')
+        # determine our orientation
+        self._affine = affine.copy()
+        codes = axcodes2ornt(aff2axcodes(self._affine))
+        self._order = np.argsort([c[0] for c in codes])
+        self._flips = np.array([c[1] < 0 for c in codes])[self._order]
+        self._flips = list(self._flips) + [False]  # add volume dim
+        self._scalers = np.abs(self._affine).max(axis=0)[:3]
```

Comment:
I added a `voxel_sizes` function in a new PR, because I keep doing the same calculation : https://github.com/nipy/nibabel/pull/413